cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/TableUpdater.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.replication;
+
+import co.cask.cdap.api.common.Bytes;
+import org.apache.hadoop.conf.Configuration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Utility class for Replication Status Tool
+ */
+public abstract class TableUpdater {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TableUpdater.class);
+  private ConcurrentHashMap<String, Long> cachedUpdates;
+  private Timer updateTimer;
+
+  public final byte[] rsID;
+  public final String columnQualifier;
+  public final Configuration conf;
+  public boolean tableExists = false;
+
+  public TableUpdater(String columnQualifier, final Configuration conf) {
+    this.columnQualifier = columnQualifier;
```

Comment:
columnQualifier is a bad name for this, it's really the pointer type (write or repl).